gateway/src/main/java/com/lyft/data/gateway/router/RoutingManager.java

```diff
@@ -65,13 +66,29 @@ public void setBackendForQueryId(String queryId, String backend) {
   }
 
   /**
-   * Performs routing.
+   * Performs routing to an adhoc backend.
    *
    * @return
    */
-  public String provideBackendForThisRequest() {
-    List<ProxyBackendConfiguration> backends = this.gatewayBackendManager.getActiveBackends();
-    int backendId = (int) (requestCounter.incrementAndGet() % backends.size());
+  public String provideAdhocBackendForThisRequest() {
+    List<ProxyBackendConfiguration> backends = this.gatewayBackendManager.getActiveAdhocBackends();
+    int backendId = (int) (requestAdhocCounter.incrementAndGet() % backends.size());
+    return backends.get(backendId).getProxyTo();
+  }
+
+  /**
+   * Performs routing to a scheduled backend. This falls back to an adhoc backend, if no scheduled
+   * backend is found.
+   *
+   * @return
+   */
+  public String provideScheduledBackendForThisRequest() {
+    List<ProxyBackendConfiguration> backends =
+        this.gatewayBackendManager.getActiveScheduledBackends();
+    if (backends.isEmpty()) {
+      backends = this.gatewayBackendManager.getActiveAdhocBackends();
```

Comment:
If we fall back to activebackend, should we use the requestAdhocCounter instead?
or can we just called the provideAdhocBackendForThisRequest method directly.